cs-ssh/src/test/java/io/cloudslang/content/ssh/services/impl/SSHServiceImplTest.java

```diff
@@ -89,25 +84,30 @@ public void testConstructors() {
-        assertEquals(sessionMock, sshService.getSSHSession());
-
-        sshService = new SSHServiceImpl(CONNECTION_DETAILS, null, new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, true);
-        assertEquals(channelExecMock, sshService.getExecChannel());
-        assertEquals(sessionMock, sshService.getSSHSession());
-
-        sshService = new SSHServiceImpl(CONNECTION_DETAILS, new KeyFile(SHELL_PATH), new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, true);
-        assertEquals(channelExecMock, sshService.getExecChannel());
-        assertEquals(sessionMock, sshService.getSSHSession());
-
-        sshService = new SSHServiceImpl(CONNECTION_DETAILS, new KeyFile(SHELL_PATH, PASS_PHRASE), new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, true);
-        assertEquals(channelExecMock, sshService.getExecChannel());
-        assertEquals(sessionMock, sshService.getSSHSession());
-
-        sshService = new SSHServiceImpl(CONNECTION_DETAILS, new KeyFile(SHELL_PATH, PASS_PHRASE), new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT);
-        assertEquals(null, sshService.getExecChannel());
-        assertEquals(sessionMock, sshService.getSSHSession());
+        ProxyHTTP proxyHTTP = null;
+        try {
+            sshService = new SSHServiceImpl(CONNECTION_DETAILS, null, new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, false, proxyHTTP, "");
+            assertEquals(null, sshService.getExecChannel());
+            assertEquals(sessionMock, sshService.getSSHSession());
+
+            sshService = new SSHServiceImpl(CONNECTION_DETAILS, null, new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, true, proxyHTTP, "");
+            assertEquals(channelExecMock, sshService.getExecChannel());
+            assertEquals(sessionMock, sshService.getSSHSession());
+
+            sshService = new SSHServiceImpl(CONNECTION_DETAILS, new KeyFile(SHELL_PATH), new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, true, proxyHTTP, "");
+            assertEquals(channelExecMock, sshService.getExecChannel());
+            assertEquals(sessionMock, sshService.getSSHSession());
+
+            sshService = new SSHServiceImpl(CONNECTION_DETAILS, new KeyFile(SHELL_PATH, PASS_PHRASE), new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, true, proxyHTTP, "");
+            assertEquals(channelExecMock, sshService.getExecChannel());
+            assertEquals(sessionMock, sshService.getSSHSession());
+
+            sshService = new SSHServiceImpl(CONNECTION_DETAILS, new KeyFile(SHELL_PATH, PASS_PHRASE), new KnownHostsFile(KNOWN_HOSTS_PATH, KNOWN_HOSTS_POLICY), CONNECT_TIMEOUT, false, proxyHTTP, "");
+            assertEquals(null, sshService.getExecChannel());
+            assertEquals(sessionMock, sshService.getSSHSession());
+        } catch (SSHException e) {
+            assert (false);
```

Comment:
The constructor now throws an exception. We have to catch the exception because of Java compiler but we don't expect one.